core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```diff
@@ -193,14 +294,192 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (nestedFilter != null) {
             builder.field("nested_filter", nestedFilter, params);
         }
-        if (coerce != null) {
-            builder.field("coerce", coerce);
-        }
-        if (ignoreMalformed != null) {
-            builder.field("ignore_malformed", ignoreMalformed);
-        }
+        builder.field("coerce", coerce);
+        builder.field("ignore_malformed", ignoreMalformed);
 
         builder.endObject();
         return builder;
     }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (! (object instanceof GeoDistanceSortBuilder)) {
```

Comment:
This made me wonder why we compare objects class in other places like this, maybe its faster? Was this a concious choice against it?